src/backend/dx11/src/lib.rs

```diff
@@ -3088,12 +3091,33 @@ pub struct InternalBuffer {
     // TODO: need to sync between `raw` and `disjoint_cb`, same way as we do with
     // `MemoryFlush/Invalidate`
     disjoint_cb: Option<*mut d3d11::ID3D11Buffer>, // if unbound this buffer might be null.
+    /// Handle to the Memory arena storing this buffer.
+    ///
+    /// The copy of the InternalBuffer inside of the arena will have an empty Weak.
+    ///
+    /// If accessed from a Buffer, once memory is bound to a buffer, this should never be None.
+    local_memory_arena: Weak<RwLock<LocalResourceArena<InternalBuffer>>>,
+    /// Index into the above memory arena.
+    ///
+    /// The copy of the InternalBuffer inside of the arena will have None.
+    ///
+    /// If accessed from a Buffer, once memory is bound to a buffer, this should never be None.
+    memory_index: Option<thunderdome::Index>,
     srv: Option<*mut d3d11::ID3D11ShaderResourceView>,
     uav: Option<*mut d3d11::ID3D11UnorderedAccessView>,
     usage: buffer::Usage,
     debug_name: Option<String>,
 }
 
+impl InternalBuffer {
+    unsafe fn release_resources(&self) {
```

Comment:
let's have it `&mut self` and zeroing out the internal fields, just for ease of debugging later on